### PR TITLE
[AIR] Remove MXNet code

### LINKED ISSUE #36909: [docs] Remove references to mxnet from Ray
### Description

MXNet doesn't seem to be seriously maintained anymore, and our docs don't build for 3.10 if we have mxnet installed.

```
ERROR: Ignored the following versions that require a different python version: 0.0.12 Requires-Python >=3.5, <3.8
ERROR: Could not find a version that satisfies the requirement mxnet==1.8.0.post0 (from versions: 1.6.0)
ERROR: No matching distribution found for mxnet==1.8.0.post0
```

### Link

_No response_

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

Development on MXNet has more or less ceased for more than a year now, so there is no point in supporting it going forward: https://github.com/apache/mxnet/commits/master

## Related issue number

Closes https://github.com/ray-project/ray/issues/36909.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/_toc.yml
Comment (matthewdeng) [inline doc/source/_toc.yml:1]: Nice! I did a quick [search](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/ray-project/ray%24+mxnet+-content:HOROVOD_WITHOUT_MXNET&patternType=standard&sm=1&groupBy=path) and it seems like there are a few more files where MXNet is being referenced. Shall we remove the rest of them in this PR in order to close https://github.com/ray-project/ray/issues/36909?

Comment (pcmoritz) [reply doc/source/_toc.yml:1]: Yeah, let me do this -- I don't think the other stuff is useful either :)

File: doc/source/ray-air/predictors.rst [hunk 1]
```diff
@@ -207,10 +207,8 @@ subclass :class:`~ray.train.predictor.Predictor` and implement:
 Examples
 ~~~~~~~~
 
-We'll walk through how to implement a predictor for two frameworks:
-
-* MXNet -- a deep learning framework like Torch.
-* statsmodel -- a Python library that provides regression and linear models.
+We'll walk through how to implement a predictor for `statsmodel` --
```
Comment (matthewdeng) [inline doc/source/ray-air/predictors.rst:210]:
```suggestion
We'll walk through how to implement a predictor for ``statsmodel`` --
```

File: doc/source/ray-air/predictors.rst [hunk 2]
```diff
@@ -222,30 +220,6 @@ Before you begin
...
-
-        Finally, create a stub for the `MXNetPredictor` class.
-
-        .. literalinclude:: doc_code/mxnet_predictor.py
-            :language: python
-            :dedent:
-            :start-after: __mxnetpredictor_signature_start__
-            :end-before: __mxnetpredictor_signature_end__
-
     .. group-tab:: statsmodel
```
Comment (matthewdeng) [inline doc/source/ray-air/predictors.rst:223]: nit: Can remove tabs now since there's only one

**Review by matthewdeng — APPROVED** [review]
Thanks!

File: doc/source/ray-air/predictors.rst (hunk 2 above)
Comment (pcmoritz) [reply doc/source/ray-air/predictors.rst:223]: I left it in to make it easier to add more examples here in the future, but happy to remove :)

**Comment by richardliaw** [comment]
awesome!

**Comment by krfricke** [comment]
I like this cleanup. Should we do the same with [GluonCV](https://github.com/dmlc/gluon-cv) (last release 2+ years ago)?

**Pull request merged**